polling_stations/apps/data_collection/management/commands/__init__.py

```diff
@@ -440,17 +461,20 @@ def add_residential_address(self, address_info):
         strip all whitespace from postcode and convert to uppercase
         this will make it easier to query this based on user-supplied postcode
         """
-        address_info['postcode'] = re.sub('[^A-Z0-9]', '', address_info['postcode'].upper())
+        address_info['postcode'] =\
+            re.sub('[^A-Z0-9]', '', address_info['postcode'].upper())
 
         # generate a unique slug so we can provide a consistent url
         slug = self.get_slug(address_info)
 
         ResidentialAddress.objects.update_or_create(
-            council=self.council,
-            address=address_info['address'],
-            postcode=address_info['postcode'],
-            polling_station_id=address_info['polling_station_id'],
-            slug=slug
+            slug=slug,
```

Comment:
This is fine, as long as we don't try and use UPRN as the slug - remember, multiple dwellings in a subdivided property share a UPRN as we discussed here: https://github.com/DemocracyClub/UK-Polling-Stations/pull/222#issuecomment-207978610